RT-tests snap (#1)
locnnil/real-time-tests-snap: README.md

# Real Time Tests Snap 

[![real-time-tests](https://snapcraft.io/real-time-tests/badge.svg)](https://snapcraft.io/real-time-tests)

[![Get it from the Snap Store](https://snapcraft.io/static/images/badges/en/snap-store-black.svg)](https://snapcraft.io/real-time-tests)

> This snap is based on the debian package
> [rt-tests](https://wiki.linuxfoundation.org/realtime/documentation/howto/tools/rt-tests)
> and it's meant to be an unofficial version of this package in snap ecosystem,
> intended for use in systems that do not have any other package manager besides
> snapd, specifically Ubuntu Core systems.

Reviewer: ```suggestion
This is the snap packaging of
[rt-tests](https://wiki.linuxfoundation.org/realtime/documentation/howto/tools/rt-tests).
```
It isn't based on the Debian package and it can be used on systems that do have snapd along with support for other package formats.

> To install this snap, it should be installed in devmode, which requires adding
> the `--devmode` flag during installation. Additionally, please note that this
> snap is only available on the unstable `--edge` channel.

Reviewer: ```suggestion
```

> This snap is based on the debian package `rt-tests` and it's meant to be an
> unofficial version of this package, meant to be used in systems that don't have
> any other package manager besides snapd, such as Ubuntu Core systems.

Reviewer: ```suggestion
```
Redundant

## Install

To install it:

```bash
sudo snap install real-time-tests
```

## Configure

> It's necessary to connect the interface [process-control](https://snapcraft.io/docs/process-control-interface) to work properlly:

Reviewer: ```suggestion
It's necessary to connect the [process-control](https://snapcraft.io/docs/process-control-interface) interface to work properly:
```

Reviewer: What does it mean to not work "properly"? Will it not work at all or fails to operate as it should? Refer to the journal or snappy-debug output to get some info on why it needs this access.

Author: In the case of `cyclictest` for example, it would fail: 

```zsh
❯ sudo real-time-tests.cyclictest --smp --priority=80 --interval=200 --distance=0
Unable to change scheduling policy!
either run as root or join realtime group
```

The snappy-debug output:

```zsh

= AppArmor =
Time: Jun 03 18:54:42
Log: apparmor="DENIED" operation="capable" class="cap" profile="/snap/snapd/21759/usr/lib/snapd/snap-confine" pid=202148 comm="snap-confine" capability=12  capname="net_admin"
Capability: net_admin
Suggestions:
* adjust program to not require 'CAP_NET_ADMIN' (see 'man 7 capabilities')
* add one of 'bluetooth-control, firewall-control, netlink-audit, netlink-connector, network-control, qualcomm-ipc-router' to 'plugs'
* do nothing if using systemd utility (eg, timedatectl): https://forum.snapcraft.io/t/managing-time-date-and-timezone-in-ubuntu-core/408/44
* do nothing (https://launchpad.net/bugs/1465724)

= AppArmor =
Time: Jun 03 18:54:42
Log: apparmor="DENIED" operation="capable" class="cap" profile="snap.real-time-tests.cyclictest" pid=202148 comm="cyclictest" capability=23  capname="sys_nice"
Capability: sys_nice
Suggestions:
* adjust program to not require 'CAP_SYS_NICE' (see 'man 7 capabilities')
* add one of 'process-control' to 'plugs'
* do nothing if program otherwise works properly
```

```bash
sudo snap connect real-time-tests:process-control :process-control
```

### Local Build

```bash
snapcraft -v

> sudo snap install --dangerous ./real-time-tests_0.1_amd64.snap

Reviewer: ```suggestion
sudo snap install --dangerous *.snap
```
